geomet/esri.py

```diff
@@ -0,0 +1,202 @@
+#  Copyright 2013 Lars Butler & individual contributors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import json
+#-------------------------------------------------------------------------
+def load(source_file):
+    """
+    Converts Esri Json File to GeoJSON.
+
+    :param source_file:
+        Path to a file that contains the Esri JSON data.
+    
+    :returns: 
+         A GeoJSON `dict` representing the geometry read from the file.
+
+    """
+    with open(source_file, 'r') as reader:
+        data = json.loads(reader.read())
+        return loads(data)
+#-------------------------------------------------------------------------
+def loads(string):
+    """
+    Construct a GeoJSON `dict` from Esri JSON (string/dict).
+
+    :param string:
+        The Esri JSON geometry representation
+
+    :returns: 
+         A GeoJSON `dict` representing the geometry read from the file.
+    """
+    if isinstance(string, str):
+        string = json.loads(string)
+    if 'rings' in string:
+        return _esri_to_geojson_convert['rings'](string)
+    elif 'paths' in string:
+        return _esri_to_geojson_convert['paths'](string)
+    elif 'x' in string or 'y' in string:
+        return _esri_to_geojson_convert['x'](string)
+    elif 'points' in string:
+        return _esri_to_geojson_convert['points'](string)
+    else:
+        raise ValueError("Invalid EsriJSON: %s" % string)
+#-------------------------------------------------------------------------
+def dump(obj, dest_file):
+    """
+    Converts GeoJSON to Esri JSON File.
+    """
+    with open(dest_file, 'w') as writer:
```

Comment:
Same thing with `load` above - `dump` should write to an open file-like object, not open the file itself.

Reply:
Comment as above except I'll use `json.dump`